### ISSUE TITLE
[React JS] fetchAuthSession is always returning undefined values

### ISSUE THREAD

**Issue by gokulakannant** [aws-amplify/amplify-js#13487]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication, Geo

### Amplify Version

v6

### Amplify Categories

auth

### Backend

None

### Environment information

<details>

```
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 1.81 GB / 15.73 GB
  Binaries:
    Node: 16.13.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (125.0.2535.92)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @aws-amplify/geo: ^3.0.35 => 3.0.35 
    @aws-amplify/geo/location-service:  undefined ()
    @aws-amplify/ui-react: ^6.1.12 => 6.1.12 
    @aws-amplify/ui-react-geo: ^2.0.16 => 2.0.16 
    @aws-amplify/ui-react-internal:  undefined ()
    @babel/plugin-proposal-nullish-coalescing-operator: ^7.18.6 => 7.18.6 
    @babel/plugin-proposal-optional-chaining: ^7.21.0 => 7.21.0 
    @material-ui/core: ^4.12.3 => 4.12.4 
    @material-ui/icons: ^4.11.2 => 4.11.3
    @material-ui/lab: ^4.0.0-alpha.60 => 4.0.0-alpha.61
    @testing-library/jest-dom: ^5.11.4 => 5.17.0
    @testing-library/react: ^11.1.0 => 11.2.7
    @testing-library/user-event: ^12.1.10 => 12.8.3
    aws-amplify: ^6.3.6 => 6.3.6
    aws-amplify/adapter-core:  undefined ()
    aws-amplify/analytics:  undefined ()
    aws-amplify/analytics/kinesis:  undefined ()
    aws-amplify/analytics/kinesis-firehose:  undefined ()
    aws-amplify/analytics/personalize:  undefined ()
    aws-amplify/analytics/pinpoint:  undefined ()
    aws-amplify/api:  undefined ()
    aws-amplify/api/server:  undefined ()
    aws-amplify/auth:  undefined ()
    aws-amplify/auth/cognito:  undefined ()
    aws-amplify/auth/cognito/server:  undefined ()
    aws-amplify/auth/enable-oauth-listener:  undefined ()
    aws-amplify/auth/server:  undefined ()
    aws-amplify/data:  undefined ()
    aws-amplify/data/server:  undefined ()
    aws-amplify/datastore:  undefined ()
    aws-amplify/in-app-messaging:  undefined ()
    aws-amplify/in-app-messaging/pinpoint:  undefined ()
    aws-amplify/push-notifications:  undefined ()
    aws-amplify/push-notifications/pinpoint:  undefined ()
    aws-amplify/storage:  undefined ()
    aws-amplify/storage/s3:  undefined ()
    aws-amplify/storage/s3/server:  undefined ()
    aws-amplify/storage/server:  undefined ()
    aws-amplify/utils:  undefined ()
    axios: ^0.21.2 => 0.21.4
    browserslist: ^4.18.1 => 4.23.1 (4.14.2)
    crypto-browserify: ^3.12.0 => 3.12.0
    customize-cra: ^1.0.0 => 1.0.0
    eslint: ^7.32.0 => 7.32.0 (6.8.0)
    eslint-config-node: ^4.1.0 => 4.1.0
    eslint-plugin-import: ^2.25.3 => 2.29.1
    eslint-plugin-react: ^7.27.0 => 7.34.2
    eslint-plugin-react-hooks: ^4.3.0 => 4.6.2
    events: ^3.3.0 => 3.3.0
    i18next: ^21.6.3 => 21.10.0
    jwt-decode: 3.1.2 => 3.1.2
    moment: ^2.29.1 => 2.30.1
    prop-types: ^15.7.2 => 15.8.1
    react: ^17.0.2 => 17.0.2
    react-app-rewired: ^2.2.1 => 2.2.1
    react-datepicker: ^4.2.1 => 4.25.0
    react-dom: ^17.0.2 => 17.0.2
    react-i18next: ^11.15.1 => 11.18.6
    react-redux: ^7.2.4 => 7.2.9
    react-router-dom: ^5.3.0 => 5.3.4
    react-scripts: ^4.0.3 => 4.0.3
    redux: ^4.1.1 => 4.2.1
    redux-devtools-extension: ^2.13.9 => 2.13.9
    redux-saga: ^1.1.3 => 1.3.0
    redux-saga/effects:  undefined ()
    sass: ^1.43.4 => 1.77.4
    web-vitals: ^1.0.1 => 1.1.2
    webpack-bundle-analyzer: ^4.5.0 => 4.10.2
  npmGlobalPackages:
    @aws-amplify/cli: 12.10.1
    create-react-app: 5.0.1
    express-generator: 4.16.1
    http-server: 14.1.1
    yarn: 1.22.17


```

</details>

### Describe the bug

The code sample is below: 

```js
import { fetchAuthSession } from "aws-amplify/auth";

const amplifyConfig = {
  Auth: {
    Cognito: {
      identityPoolId: IDENTITY_POOL_ID, // REQUIRED - Amazon Cognito Identity Pool ID
      allowGuestAccess: true,
    },
  },
  Geo: {
    LocationService: {
      maps: {
        items: {
          [MAP.NAME]: {
            // REQUIRED - Amazon Location Service Map resource name
            style: MAP.STYLE, // REQUIRED - String representing the style of map resource
          },
        },
        default: MAP.NAME, // REQUIRED - Amazon Location Service Map resource name to set as default
      },
      // Amplify.configure requires searchIndices, while LocationSearch requires search_indices
      searchIndices: {
        items: [PLACE], // REQUIRED - Amazon Location Service Place Index name
        default: PLACE, // REQUIRED - Amazon Location Service Place Index name to set as default
      },
      search_indices: {
        items: [PLACE], // REQUIRED - Amazon Location Service Place Index name
        default: PLACE, // REQUIRED - Amazon Location Service Place Index name to set as default
      },
      region: REGION, // REQUIRED - Amazon Location Service Region
    },
  },
}
Amplify.configure(amplifyConfig);


 //I have called the fetchAuthSession method inside the component
  useEffect(() => {
    const fetchCredentials = async () => {
      // Fetch AWS credentials from Amazon Cognito using Amplify Auth and storing it in state
      **const authSession = await fetchAuthSession();**
      console.log("fetchAuthSession", authSession);
      setCredentials(authSession.credentials);
    };
    fetchCredentials();
  }, []);

```

fetchAuthSession always returning the below output,

_{tokens: undefined, credentials: undefined, identityId: undefined, userSub: undefined}_

<img width="959" alt="image" src="https://github.com/aws-amplify/amplify-js/assets/38354007/c3719bc8-4b76-4063-ace7-458611fddb51">

Also, I don't see any cognito network call in the browser network tab.

I tried to delete my application package-lock.json and reinstall the packages again. But the result is the same. 

It doesn't throw any error in the console.

### Expected behavior

The expected behaviour is fetchAuthSession should return the credentials

### Reproduction steps

Always it is returning the empty values (_{tokens: undefined, credentials: undefined, identityId: undefined, userSub: undefined}_). 

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### Manual configuration

```json

{
  Auth: {
    Cognito: {
      identityPoolId: IDENTITY_POOL_ID, // REQUIRED - Amazon Cognito Identity Pool ID
      allowGuestAccess: true,
    },
  },
  Geo: {
    LocationService: {
      maps: {
        items: {
          [MAP.NAME]: {
            // REQUIRED - Amazon Location Service Map resource name
            style: MAP.STYLE, // REQUIRED - String representing the style of map resource
          },
        },
        default: MAP.NAME, // REQUIRED - Amazon Location Service Map resource name to set as default
      },
      // Amplify.configure requires searchIndices, while LocationSearch requires search_indices
      searchIndices: {
        items: [PLACE], // REQUIRED - Amazon Location Service Place Index name
        default: PLACE, // REQUIRED - Amazon Location Service Place Index name to set as default
      },
      search_indices: {
        items: [PLACE], // REQUIRED - Amazon Location Service Place Index name
        default: PLACE, // REQUIRED - Amazon Location Service Place Index name to set as default
      },
      region: REGION, // REQUIRED - Amazon Location Service Region
    },
  },
}

````

**Comment by cwomack** [comment]
Hello, @gokulakannant and thanks for opening this issue.  Did this only start happening recently after upgrading to a more recent version?  

Related to #13486

**Comment by ashika112** [comment]
Hi can you provide some additional info on how you are setting up your backend? I was trying to reproduce this and im unable too. 

My configure looks like 

```
Amplify.configure({
  Auth: {
    Cognito: {
      allowGuestAccess: true,
      identityPoolId: "XXXXXX",
    },
  },
});
```
My call of `fetchAuthSession` in useEffect looks like the following
 ```
 useEffect(() => {
    fetchAuthSession().then((r) => {
      console.log("fetchAuth", r);
      const sub = r.userSub ?? "no user";
      setUser(sub);
    }); // will return the credentials
  }, []);
  ```

**Comment by gokulakannant** [comment]
@ashika112 , I manually created the backend setup in aws console. When I call the "fetchAuthSession()" function, the cognito network call is not initiated. I think, that is the root cause of the issue.

When I tried the same credentials with the repo ([amazon-location-amplify-ui-geo-explore](https://github.com/aws-samples/amazon-location-amplify-ui-geo-explore)), it works. I could see the cognito network request (https://cognito-identity.eu-west-1.amazonaws.com/) in the Network tab

**Comment by ashika112** [comment]
Are you using the same backend in the sample app you linked as well? I am not sure why it would be `undefined` i would compare against the sample app to see if there is mismatch to your app.

**Comment by gokulakannant** [comment]
Hello Team,

As you see in the Environment details, I am using "react-scripts: ^4.0.3 => 4.0.3", it doesn't compile the .mjs files. So, I tried to override the webpack configuration by installing "react-app-rewired". On top of it, I am using aws-amplify. As some of its dependencies have .mjs files, it was compiled by webpack overridden configuration. It has some issues, so the fetchAuthSession is not working.

Then I upgraded the react-scripts to 5.x.x. Then it works.

So it seems this issue is not related to aws-amplify. Thank you for the support...!!
